### ISSUE TITLE
Wrong server loader

### ISSUE THREAD

**Issue by kreezxil** [Griefed/ServerPackCreator#69]
I told it to make a 1.16.5 Fabric 0.11.6 server, but it made a 1.17.1 Fabric 0.11.6 server.
```md
#modloader log
2021-07-20T16:24:28,287  INFO [pool-2-thread-1] (CreateServerPack.java:569) - Starting Fabric installation.
2021-07-20T16:24:28,532  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Loading Fabric Installer: 0.7.4
2021-07-20T16:24:28,610  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Using latest loader version
2021-07-20T16:24:29,560  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Using latest game version
2021-07-20T16:24:30,034  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Installing Fabric Loader 0.11.6(1.17.1) on the server
2021-07-20T16:24:30,037  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading required files
2021-07-20T16:24:30,074  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading library net.fabricmc:tiny-mappings-parser:0.2.2.14
2021-07-20T16:24:30,300  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading library net.fabricmc:sponge-mixin:0.9.4+mixin.0.8.2
2021-07-20T16:24:30,392  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading library net.fabricmc:tiny-remapper:0.4.2
2021-07-20T16:24:30,417  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading library net.fabricmc:access-widener:1.0.0
2021-07-20T16:24:30,444  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading library net.fabricmc:fabric-loader-sat4j:2.3.5.4
2021-07-20T16:24:30,497  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading library com.google.jimfs:jimfs:1.2-fabric
2021-07-20T16:24:30,533  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading library org.ow2.asm:asm:9.1
2021-07-20T16:24:30,567  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading library org.ow2.asm:asm-analysis:9.1
2021-07-20T16:24:30,592  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading library org.ow2.asm:asm-commons:9.1
2021-07-20T16:24:30,620  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading library org.ow2.asm:asm-tree:9.1
2021-07-20T16:24:30,646  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading library org.ow2.asm:asm-util:9.1
2021-07-20T16:24:30,673  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading library net.fabricmc:intermediary:1.17.1
2021-07-20T16:24:30,741  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading library net.fabricmc:fabric-loader:0.11.6
2021-07-20T16:24:30,798  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading library com.google.guava:guava:21.0
2021-07-20T16:24:31,016  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR
2021-07-20T16:24:31,020  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR: tiny-mappings-parser-0.2.2.14.jar
2021-07-20T16:24:31,072  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR: sponge-mixin-0.9.4+mixin.0.8.2.jar
2021-07-20T16:24:31,306  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR: tiny-remapper-0.4.2.jar
2021-07-20T16:24:31,325  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR: access-widener-1.0.0.jar
2021-07-20T16:24:31,329  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR: fabric-loader-sat4j-2.3.5.4.jar
2021-07-20T16:24:31,401  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR: jimfs-1.2-fabric.jar
2021-07-20T16:24:31,446  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR: asm-9.1.jar
2021-07-20T16:24:31,466  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR: asm-analysis-9.1.jar
2021-07-20T16:24:31,466  INFO [pool-2-thread-1] (CreateServerPack.java:587) - duplicate file: module-info.class
2021-07-20T16:24:31,472  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR: asm-commons-9.1.jar
2021-07-20T16:24:31,472  INFO [pool-2-thread-1] (CreateServerPack.java:587) - duplicate file: module-info.class
2021-07-20T16:24:31,485  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR: asm-tree-9.1.jar
2021-07-20T16:24:31,485  INFO [pool-2-thread-1] (CreateServerPack.java:587) - duplicate file: module-info.class
2021-07-20T16:24:31,498  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR: asm-util-9.1.jar
2021-07-20T16:24:31,499  INFO [pool-2-thread-1] (CreateServerPack.java:587) - duplicate file: module-info.class
2021-07-20T16:24:31,512  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR: intermediary-1.17.1.jar
2021-07-20T16:24:31,627  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR: fabric-loader-0.11.6.jar
2021-07-20T16:24:31,709  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Generating server launch JAR: guava-21.0.jar
2021-07-20T16:24:32,981  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Done, start server by running fabric-server-launch.jar
2021-07-20T16:24:32,981  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Downloading Minecraft server
2021-07-20T16:24:36,712  INFO [pool-2-thread-1] (CreateServerPack.java:587) - Done
2021-07-20T16:24:36,733  INFO [pool-2-thread-1] (CreateServerPack.java:589) - Returning to ServerPackCreator.
```

```md
#serverpack creator log
2021-07-20T16:24:27,144  INFO [AWT-EventQueue-0] (CreateServerPackTab.java:702) - Checking entered configuration.
2021-07-20T16:24:27,151  INFO [AWT-EventQueue-0] (Configuration.java:1971) - Successfully written new configuration file.
2021-07-20T16:24:27,151  INFO [AWT-EventQueue-0] (Configuration.java:583) - Checking configuration...
2021-07-20T16:24:27,157  WARN [AWT-EventQueue-0] (Configuration.java:591) - No clientside-only mods specified. Using fallback list.
2021-07-20T16:24:27,161 ERROR [AWT-EventQueue-0] (Configuration.java:1192) - Incorrect Java path specified.
2021-07-20T16:24:27,162  WARN [AWT-EventQueue-0] (Configuration.java:1157) - You didn't specify the path to your Java installation. ServerPackCreator will try to determine it for you...
2021-07-20T16:24:27,163  WARN [AWT-EventQueue-0] (Configuration.java:1165) - ServerPackCreator set the path to your Java installation to: /usr/lib/jvm/java-8-openjdk/jre/bin/java
2021-07-20T16:24:28,051  INFO [AWT-EventQueue-0] (Configuration.java:1045) - Your configuration is:
2021-07-20T16:24:28,051  INFO [AWT-EventQueue-0] (Configuration.java:1046) - Modpack directory: /home/kreezxil/.config/gdlauncher_next/instances/1.16.5 Fabric
2021-07-20T16:24:28,052  INFO [AWT-EventQueue-0] (Configuration.java:1052) - Client mods specified. Client mods are:
2021-07-20T16:24:28,052  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     AmbientSounds
2021-07-20T16:24:28,052  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     BackTools
2021-07-20T16:24:28,053  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     BetterAdvancement
2021-07-20T16:24:28,053  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     BetterPing
2021-07-20T16:24:28,053  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     cherished
2021-07-20T16:24:28,054  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     ClientTweaks
2021-07-20T16:24:28,054  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     Controlling
2021-07-20T16:24:28,054  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     DefaultOptions
2021-07-20T16:24:28,054  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     durability
2021-07-20T16:24:28,055  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     DynamicSurroundings
2021-07-20T16:24:28,055  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     itemzoom
2021-07-20T16:24:28,055  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     jei-professions
2021-07-20T16:24:28,055  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     jeiintegration
2021-07-20T16:24:28,055  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     JustEnoughResources
2021-07-20T16:24:28,056  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     MouseTweaks
2021-07-20T16:24:28,056  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     Neat
2021-07-20T16:24:28,056  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     OldJavaWarning
2021-07-20T16:24:28,057  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     PackMenu
2021-07-20T16:24:28,057  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     preciseblockplacing
2021-07-20T16:24:28,057  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     SimpleDiscordRichPresence
2021-07-20T16:24:28,057  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     SpawnerFix
2021-07-20T16:24:28,057  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     TipTheScales
2021-07-20T16:24:28,058  INFO [AWT-EventQueue-0] (Configuration.java:1054) -     WorldNameRandomizer
2021-07-20T16:24:28,058  INFO [AWT-EventQueue-0] (Configuration.java:1059) - Directories to copy:
2021-07-20T16:24:28,058  INFO [AWT-EventQueue-0] (Configuration.java:1064) -     config
2021-07-20T16:24:28,058  INFO [AWT-EventQueue-0] (Configuration.java:1064) -     mods
2021-07-20T16:24:28,059  INFO [AWT-EventQueue-0] (Configuration.java:1071) - Include server installation:      true
2021-07-20T16:24:28,059  INFO [AWT-EventQueue-0] (Configuration.java:1072) - Java Installation path:           /usr/lib/jvm/java-8-openjdk/jre/bin/java
2021-07-20T16:24:28,059  INFO [AWT-EventQueue-0] (Configuration.java:1073) - Minecraft version:                1.16.5
2021-07-20T16:24:28,059  INFO [AWT-EventQueue-0] (Configuration.java:1074) - Modloader:                        Fabric
2021-07-20T16:24:28,060  INFO [AWT-EventQueue-0] (Configuration.java:1075) - Modloader Version:                0.11.6
2021-07-20T16:24:28,060  INFO [AWT-EventQueue-0] (Configuration.java:1076) - Include server icon:              true
2021-07-20T16:24:28,060  INFO [AWT-EventQueue-0] (Configuration.java:1077) - Include server properties:        true
2021-07-20T16:24:28,060  INFO [AWT-EventQueue-0] (Configuration.java:1078) - Include start scripts:            true
2021-07-20T16:24:28,061  INFO [AWT-EventQueue-0] (Configuration.java:1079) - Create zip-archive of serverpack: true
2021-07-20T16:24:28,061  INFO [AWT-EventQueue-0] (Configuration.java:628) - Config check successful. No errors encountered.
2021-07-20T16:24:28,061  INFO [AWT-EventQueue-0] (CreateServerPackTab.java:722) - Configuration checked successfully.
2021-07-20T16:24:28,062  INFO [AWT-EventQueue-0] (CreateServerPackTab.java:729) - Deleted temporary config file.
2021-07-20T16:24:28,062  INFO [AWT-EventQueue-0] (CreateServerPackTab.java:736) - Writing GUI configuration to new serverpackcreator.conf.
2021-07-20T16:24:28,063  INFO [AWT-EventQueue-0] (Configuration.java:1951) - Deleted existing configuration file.
2021-07-20T16:24:28,063  INFO [AWT-EventQueue-0] (Configuration.java:1971) - Successfully written new configuration file.
2021-07-20T16:24:28,064  INFO [AWT-EventQueue-0] (CreateServerPackTab.java:755) - Starting ServerPackCreator run.
2021-07-20T16:24:28,069  INFO [pool-2-thread-1] (Configuration.java:583) - Checking configuration...
2021-07-20T16:24:28,079  WARN [pool-2-thread-1] (Configuration.java:591) - No clientside-only mods specified. Using fallback list.
2021-07-20T16:24:28,082 ERROR [pool-2-thread-1] (Configuration.java:1192) - Incorrect Java path specified.
2021-07-20T16:24:28,083  WARN [pool-2-thread-1] (Configuration.java:1157) - You didn't specify the path to your Java installation. ServerPackCreator will try to determine it for you...
2021-07-20T16:24:28,084  WARN [pool-2-thread-1] (Configuration.java:1165) - ServerPackCreator set the path to your Java installation to: /usr/lib/jvm/java-8-openjdk/jre/bin/java
2021-07-20T16:24:28,195  INFO [pool-2-thread-1] (Configuration.java:1045) - Your configuration is:
2021-07-20T16:24:28,195  INFO [pool-2-thread-1] (Configuration.java:1046) - Modpack directory: /home/kreezxil/.config/gdlauncher_next/instances/1.16.5 Fabric
2021-07-20T16:24:28,195  INFO [pool-2-thread-1] (Configuration.java:1052) - Client mods specified. Client mods are:
2021-07-20T16:24:28,195  INFO [pool-2-thread-1] (Configuration.java:1054) -     AmbientSounds
2021-07-20T16:24:28,196  INFO [pool-2-thread-1] (Configuration.java:1054) -     BackTools
2021-07-20T16:24:28,196  INFO [pool-2-thread-1] (Configuration.java:1054) -     BetterAdvancement
2021-07-20T16:24:28,196  INFO [pool-2-thread-1] (Configuration.java:1054) -     BetterPing
2021-07-20T16:24:28,196  INFO [pool-2-thread-1] (Configuration.java:1054) -     cherished
2021-07-20T16:24:28,196  INFO [pool-2-thread-1] (Configuration.java:1054) -     ClientTweaks
2021-07-20T16:24:28,196  INFO [pool-2-thread-1] (Configuration.java:1054) -     Controlling
2021-07-20T16:24:28,197  INFO [pool-2-thread-1] (Configuration.java:1054) -     DefaultOptions
2021-07-20T16:24:28,197  INFO [pool-2-thread-1] (Configuration.java:1054) -     durability
2021-07-20T16:24:28,197  INFO [pool-2-thread-1] (Configuration.java:1054) -     DynamicSurroundings
2021-07-20T16:24:28,197  INFO [pool-2-thread-1] (Configuration.java:1054) -     itemzoom
2021-07-20T16:24:28,197  INFO [pool-2-thread-1] (Configuration.java:1054) -     jei-professions
2021-07-20T16:24:28,198  INFO [pool-2-thread-1] (Configuration.java:1054) -     jeiintegration
2021-07-20T16:24:28,198  INFO [pool-2-thread-1] (Configuration.java:1054) -     JustEnoughResources
2021-07-20T16:24:28,198  INFO [pool-2-thread-1] (Configuration.java:1054) -     MouseTweaks
2021-07-20T16:24:28,198  INFO [pool-2-thread-1] (Configuration.java:1054) -     Neat
2021-07-20T16:24:28,198  INFO [pool-2-thread-1] (Configuration.java:1054) -     OldJavaWarning
2021-07-20T16:24:28,198  INFO [pool-2-thread-1] (Configuration.java:1054) -     PackMenu
2021-07-20T16:24:28,199  INFO [pool-2-thread-1] (Configuration.java:1054) -     preciseblockplacing
2021-07-20T16:24:28,199  INFO [pool-2-thread-1] (Configuration.java:1054) -     SimpleDiscordRichPresence
2021-07-20T16:24:28,199  INFO [pool-2-thread-1] (Configuration.java:1054) -     SpawnerFix
2021-07-20T16:24:28,199  INFO [pool-2-thread-1] (Configuration.java:1054) -     TipTheScales
2021-07-20T16:24:28,199  INFO [pool-2-thread-1] (Configuration.java:1054) -     WorldNameRandomizer
2021-07-20T16:24:28,199  INFO [pool-2-thread-1] (Configuration.java:1059) - Directories to copy:
2021-07-20T16:24:28,200  INFO [pool-2-thread-1] (Configuration.java:1064) -     config
2021-07-20T16:24:28,200  INFO [pool-2-thread-1] (Configuration.java:1064) -     mods
2021-07-20T16:24:28,200  INFO [pool-2-thread-1] (Configuration.java:1071) - Include server installation:      true
2021-07-20T16:24:28,200  INFO [pool-2-thread-1] (Configuration.java:1072) - Java Installation path:           /usr/lib/jvm/java-8-openjdk/jre/bin/java
2021-07-20T16:24:28,200  INFO [pool-2-thread-1] (Configuration.java:1073) - Minecraft version:                1.16.5
2021-07-20T16:24:28,201  INFO [pool-2-thread-1] (Configuration.java:1074) - Modloader:                        Fabric
2021-07-20T16:24:28,201  INFO [pool-2-thread-1] (Configuration.java:1075) - Modloader Version:                0.11.6
2021-07-20T16:24:28,201  INFO [pool-2-thread-1] (Configuration.java:1076) - Include server icon:              true
2021-07-20T16:24:28,201  INFO [pool-2-thread-1] (Configuration.java:1077) - Include server properties:        true
2021-07-20T16:24:28,201  INFO [pool-2-thread-1] (Configuration.java:1078) - Include start scripts:            true
2021-07-20T16:24:28,201  INFO [pool-2-thread-1] (Configuration.java:1079) - Create zip-archive of serverpack: true
2021-07-20T16:24:28,202  INFO [pool-2-thread-1] (Configuration.java:628) - Config check successful. No errors encountered.
2021-07-20T16:24:28,202  INFO [pool-2-thread-1] (CreateServerPack.java:278) - Found old server_pack. Cleaning up...
2021-07-20T16:24:28,237  INFO [pool-2-thread-1] (CreateServerPack.java:300) - Cleanup of previous server_pack completed.
2021-07-20T16:24:28,237  INFO [pool-2-thread-1] (CreateServerPack.java:304) - Found old server_pack.zip. Cleaning up...
2021-07-20T16:24:28,240  INFO [pool-2-thread-1] (CreateServerPack.java:307) - Old server_pack.zip deleted.
2021-07-20T16:24:28,241  INFO [pool-2-thread-1] (CreateServerPack.java:381) - Setting up /home/kreezxil/.config/gdlauncher_next/instances/1.16.5 Fabric/server_pack/config files.
2021-07-20T16:24:28,263  INFO [pool-2-thread-1] (CreateServerPack.java:381) - Setting up /home/kreezxil/.config/gdlauncher_next/instances/1.16.5 Fabric/server_pack/mods files.
2021-07-20T16:24:28,264  INFO [pool-2-thread-1] (CreateServerPack.java:482) - Preparing a list of mods to include in server pack...
2021-07-20T16:24:28,286  INFO [pool-2-thread-1] (CreateServerPack.java:338) - Copying Fabric start scripts...
2021-07-20T16:24:28,287  INFO [pool-2-thread-1] (CreateServerPack.java:568) - Starting Fabric installation.
2021-07-20T16:24:28,287  INFO [pool-2-thread-1] (CreateServerPack.java:824) - Trying to download Fabric installer...
2021-07-20T16:24:28,324  INFO [pool-2-thread-1] (CreateServerPack.java:883) - Successfully retrieved Fabric-Installer XML.
2021-07-20T16:24:28,355  INFO [pool-2-thread-1] (CreateServerPack.java:571) - Fabric installer successfully downloaded. Installing Fabric. This may take a while...
2021-07-20T16:24:36,734  INFO [pool-2-thread-1] (CreateServerPack.java:591) - For details regarding the installation of this modloader server, see logs/modloader_installer.log.
2021-07-20T16:24:36,735  INFO [pool-2-thread-1] (CreateServerPack.java:592) - Returning to ServerPackCreator.
2021-07-20T16:24:36,987  INFO [pool-2-thread-1] (CreateServerPack.java:727) - Fabric shell script generated.
2021-07-20T16:24:36,988  INFO [pool-2-thread-1] (CreateServerPack.java:755) - Fabric batch script generated.
2021-07-20T16:24:36,988  INFO [pool-2-thread-1] (CreateServerPack.java:938) - Cleanup after modloader server installation.
2021-07-20T16:24:36,989  INFO [pool-2-thread-1] (CreateServerPack.java:946) - Deleted fabric-installer.xml
2021-07-20T16:24:36,989  INFO [pool-2-thread-1] (CreateServerPack.java:951) - Deleted fabric-installer.jar
2021-07-20T16:24:36,989  INFO [pool-2-thread-1] (CreateServerPack.java:514) - Copying server-icon.png...
2021-07-20T16:24:36,990  INFO [pool-2-thread-1] (CreateServerPack.java:533) - Copying server.properties...
2021-07-20T16:24:36,990  INFO [pool-2-thread-1] (CreateServerPack.java:653) - Creating zip archive of serverpack...
2021-07-20T16:24:37,707  WARN [pool-2-thread-1] (CreateServerPack.java:671) - !!!       NOTE: The minecraft_server.jar will not be included in the zip-archive.       !!!
2021-07-20T16:24:37,707  WARN [pool-2-thread-1] (CreateServerPack.java:672) - !!! Mojang strictly prohibits the distribution of their software through third parties. !!!
2021-07-20T16:24:37,707  WARN [pool-2-thread-1] (CreateServerPack.java:673) - !!!   Tell your users to execute the download scripts to get the Minecraft server jar.  !!!
2021-07-20T16:24:37,707  INFO [pool-2-thread-1] (CreateServerPack.java:675) - Finished creation of zip archive.
2021-07-20T16:24:37,708  INFO [pool-2-thread-1] (CreateServerPack.java:259) - Server pack available at: /home/kreezxil/.config/gdlauncher_next/instances/1.16.5 Fabric/server_pack
2021-07-20T16:24:37,708  INFO [pool-2-thread-1] (CreateServerPack.java:260) - Server pack archive available at: /home/kreezxil/.config/gdlauncher_next/instances/1.16.5 Fabric/server_pack.zip
2021-07-20T16:24:37,708  INFO [pool-2-thread-1] (CreateServerPack.java:261) - Done!
```

**Comment by Griefed** [comment]
Thanks for the report.

I'll take a look later today as I am currently at work. What I can tell you so far is that the first log you provided is from the modloader_installer log and therefore not part of ServerPackCreator code itself. `CreateServerPack.java:587` refers to the [process builder](https://github.com/Griefed/ServerPackCreator/blob/main/src/main/java/de/griefed/serverpackcreator/CreateServerPack.java#L587) which runs the modloader installer.

Fabric 0.11.6 seems to support both [1.16.5](https://fabricmc.net/versions.html?&version=1.16.5) AND [1.17.x](https://fabricmc.net/versions.html?&version=1.17.1). Will have to check what I can do, but unfortunately a fix will have to wait >9 hours.

**Comment by Griefed** [comment]
Strange.
From Fabric installation docs https://fabricmc.net/wiki/install
> Available options
    -snapshot Enables the usage of snapshot versions of Minecraft.
    -dir Used to select the installation dir, defaults to the current working directory.
    -mcversion Used to select the minecraft version, defaults to the latest stable version.
    -loader Used to select the loader version, defaults to the latest.
    -downloadMinecraft Used to automatically download the Minecraft server jar
    -noprofile Skip creating client profile
    -mavenurl (advanced) Use a custom maven url when installing
    -metaurl (advanced) Use a custom meta server url

And the ProcessBuilder for Fabric:
```java
ProcessBuilder processBuilder = new ProcessBuilder(
                            javaPath,
                            "-jar",
                            "fabric-installer.jar",
                            "server",
                            String.format("-mcversion %s", minecraftVersion),
                            String.format("-loader %s", modLoaderVersion),
                            "-downloadMinecraft").directory(new File(String.format("%s/server_pack", modpackDir)));
```

Hohumm. Will investigate more later.

**Comment by Griefed** [comment]
Was able to reproduce this with version 2.1.0. Investigating cause of issue.

**Comment by Griefed** [comment]
Fixed in version 2.1.1

The problem was a wrongly build ProcessBuilder not passing the users configuration to said ProcessBuilder correctly. Vital information about the Minecraft and Fabric Loader versions went ***poof*** due to me not setting up the ProcessBuilder in the way it was supposed to. This has been fixed and the Fabric server is now installing for the correct loader and Minecraft versions.

Before: I passed my arguments as String and formatted Strings.

Now: I'm passing a String List which contains all the vital information needed.

Closing this issue.
